### DISCUSSION TITLE
Uppercase on text input

### DISCUSSION THREAD

**Discussion by betta73** [formkit/formkit#1330] [General]
Is there a way to transform user text in uppercase?

**Comment by fenilli** [comment]
You can by using hooks, those are functions that run just before the input is added.

```ts
node.hook.commit((payload, next) => {
  return next(payload.toUpperCase())
})
```

**Reply by yogendrarr** [reply]
not working can u give me  an example for this how it will work?

**Reply by sashamilenkovic** [reply]
@yogendrarr I have an example of this here: https://github.com/orgs/formkit/discussions/1469
